common/sonar.js

```diff
@@ -11,11 +11,20 @@ function sonarStatusReferesher() {
 
     var rnd = Math.round(Math.random() * 1000000000000);
     var marker = pointURI.indexOf("&rnd=");
+    var wasPointRefreshedEarlier = marker !== -1
 
-    if (marker == -1) {
-      point.src = point.src + "&rnd=" + rnd.toString(26);
-    } else {
+    if (wasPointRefreshedEarlier) {
       point.src = point.src.substr(0, marker) + "&rnd=" + rnd.toString(26);
+    } else {
+      point.addEventListener("error", function() {
+        this.style.visibility = "hidden";
```

Comment:
The implementation is simple. 

When the script is fired for the first time (which is detected by absence of `rnd` param in the `src`), it adds listeners `error`/`load` to every point it updates. Then, on every `error` event the point that triggered such event is hidden. On every `load` even—visible.

Reply:
Pretty nice implementation btw.